src/Raven.Server/Documents/Indexes/Static/CurrentIndexingScope.cs

```diff
@@ -246,6 +251,24 @@ public unsafe dynamic LoadDocument(LazyStringValue keyLazy, string keyString, st
             }
         }
 
+        private Slice GetIdSlice(LazyStringValue id)
+        {
+            Slice idSlice;
+            if (UseNormalizedIds)
+            {
+                // making sure that we normalize the case of the id so we'll be able to find
+                // it in case-insensitive manner
+                // in addition, special characters need to be escaped
+                DocumentIdWorker.GetSliceFromId(QueryContext.Documents, id, out idSlice);
```

Comment:
Just to make sure, you want here to get _lowercased_ ID so it will be consistent with a tombstone ID hence `CleanupDocuments` will handle the deletions correctly, right? 

Will `DocumentIdWorker.GetSliceFromId()`?  I thought that only `DocumentIdWorker.GetLower()` does it.

Reply:
> in addition, special characters need to be escaped

We're aware that some inconsistencies around that possible. Please also take a look at comments in https://issues.hibernatingrhinos.com/issue/RavenDB-22116/Corrupted-document-ID-from-the-client-might-affect-the-server to double check it this won't be a potential source of other issues

CC @karmeli87

Reply:
This is currently what we use for the key.
https://github.com/ravendb/ravendb/blob/fc318e40649881a130450611c98d1417a5eaf019/src/Raven.Server/Documents/Indexes/Static/CurrentIndexingScope.cs#L312

Reply:
`GetSliceFromId` is returning a lower case `Slice`.
This is also the method that we use when creating a tombstone so it's consistent with the tombstone ID.